### ISSUE TITLE
[Question] Getting value(s) from parsed STIX 2.x Indicator object (examples included)

### ISSUE THREAD

**Issue by r3search3r** [oasis-open/cti-python-stix2#547]
Greetings,

I'm currently making an program which is parsing some cyber threat info into STIX 2.0 (or STIX 2.1) Indicator format.

I could get from some cyber threat info successfully, but I could not find any functions to get exact value from parsed STIX 2.x Indicator patterns.

So questions are...

1. Do I make function to get value from STIX 2.0 Indicator pattrens?
    (e.g. getting IP address from pattern like `ipv4-addr:value='1.2.3.4'` )
2. Or, Have you made any kind of function(s) to get value from STIX 2.0 Indicator pattrens?

Please let me know about answers above.

Sincerely Yours.

**Comment by chisholm** [comment]
This library can produce an AST from a STIX pattern (see `stix2.pattern_visitor.create_pattern_object()`).  That's just a data structure, but you could write your own code to pull things out of it, e.g. `'1.2.3.4'` would be a `StringConstant`.

Another possibility might be the stix2-patterns library (which is actually from the [cti-pattern-validator](https://github.com/oasis-open/cti-pattern-validator/) repo).  You would create a `Pattern` object from the pattern and then call `inspect()` on it.  That produces another type of data structure, which may be a bit simpler to work with.  Its downside is that that data structure doesn't preserve all aspects of the pattern like an AST does, but depending on what parts of the pattern you are interested in that may not matter.

In summary, no, there isn't a simple function to give you exactly what you want, but there are tools are available, and it shouldn't be too hard.

**Comment by rpiazza** [comment]
@r3search3r - Look in patterns.py for the AST object model...

**Comment by r3search3r** [comment]
Thank you everyone, I'll have a look at it. Will reopen this issue when I have problems or something.
